### DISCUSSION TITLE
Argparse with 'mpremote run <xxx>.py arg1 arg2' : OK on unix port , no succcess with ESP32 port ; one trick ?

### DISCUSSION THREAD

**Discussion by picpic020960** [micropython/micropython#15635] [Libraries & Drivers] [answered]
Bonjour ,
I write tool for dump I2C memory. First I test with unix port without I2C , with parameters and mpremote run vidage.py 400    ; 400 = bytes to dump . Works fine.
Then with ESP32 port and the same command , no effect.
I made wrong think but where ?
Some examples help me , thank's

PS: tryed 'input' : not better 
       Exaples found but for python , not micropython mpremote

**Comment by shariltumin** [comment, marked as answer]
Suppose you have a script "test.py" in the "/tmp" directory, as shown below:

```py
import sys
print(sys.platform)
print(arg1, arg2, arg1+arg2)

```

Doing this in a terminal in the "/tmp" directory:

```
$ cd /tmp
$ mpremote u0 mount . exec "arg1=1; arg2=3" run test.py
Local directory . is mounted at /remote
esp32
1 3 4
```

will mount the current working directory ("/tmp"), set the arg1 and arg2 values using ```exec```, load and run the "test.py" script from RAM on your esp32 board.

Another alternative, which I prefer is to create two scripts in e.g. "/tmp" directory:

```py
# args.py
arg1=1
arg2=3
```

and

```py
#test1.py
from args import *
import sys
print(sys.platform)
print(arg1, arg2, arg1+arg2)

```

Then, from a terminal in any directory on your PC, run the following:
```
$ mpremote u0 mount "/tmp" exec "import test1"
Local directory /tmp is mounted at /remote
esp32
1 3 4
```
 
Hope this helps.

**Reply by picpic020960** [reply]
Exec 'arg1=1...' is OK with no mount
Thank's

**Comment by scruss** [comment]
`mpremote run` doesn't pass command line arguments to the microcontroller board

**Reply by picpic020960** [reply]
So why argparse in micropython lib?

**Reply by shariltumin** [reply]
I do not know why, I never use it. Maybe it is just for the Unix port. I haven't seen "argparse" in the esp32 port. Maybe it makes no sense on any MCU since we go straight to REPL or run main.py if it exists.

**Reply by andrewleech** [reply]
argparse is regularly used on the Unix port, for instance it's a dependency of the command line interface to unittest used there. 

The embedded ports don't have the `sys.argv` and related parts of command line arguments though.
